src/rpc/ngHandlers/ServerInfo.h

```diff
@@ -0,0 +1,237 @@
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <backend/BackendInterface.h>
+#include <backend/DBHelpers.h>
+#include <main/Build.h>
+#include <rpc/Errors.h>
+#include <rpc/RPCHelpers.h>
+#include <rpc/common/Types.h>
+#include <rpc/common/Validators.h>
+
+#include <fmt/core.h>
+
+class SubscriptionManager;
+class ReportingETL;
+class ETLLoadBalancer;
+
+namespace RPC {
+class Counters;
+}
+
+namespace RPCng {
+
+template <
+    typename SubscriptionManagerType,
+    typename ETLLoadBalancerType,
+    typename ReportingETLType,
+    typename CountersType>
+class BaseServerInfoHandler
+{
+    std::shared_ptr<BackendInterface> backend_;
+    std::shared_ptr<SubscriptionManagerType> subscriptions_;
+    std::shared_ptr<ETLLoadBalancerType> balancer_;
+    std::shared_ptr<ReportingETLType const> etl_;
+    std::reference_wrapper<CountersType const> counters_;
+
+public:
+    struct AdminSection
+    {
+        boost::json::object counters = {};
+        boost::json::object subscriptions = {};
+        boost::json::object etl = {};
+    };
+
+    struct ValidatedLedgerSection
+    {
+        uint32_t age = 0;
+        std::string hash = {};
+        ripple::LedgerIndex seq = {};
+        std::optional<ripple::Fees> fees = std::nullopt;
+    };
+
+    struct CacheSection
+    {
+        std::size_t size = 0;
+        bool isFull = false;
+        ripple::LedgerIndex latestLedgerSeq = {};
+        float objectHitRate = 1.0;
+        float successorHitRate = 1.0;
+    };
+
+    struct InfoSection
+    {
+        std::optional<AdminSection> adminSection = std::nullopt;
+        std::string completeLedgers = {};
+        uint32_t loadFactor = 1u;
+        std::string clioVersion = Build::getClioVersionString();
+        std::optional<boost::json::object> rippledInfo = std::nullopt;
+        ValidatedLedgerSection validatedLedger = {};
+        CacheSection cache = {};
+    };
+
+    struct Output
+    {
+        InfoSection info = {};
+
+        // validated should be sent via framework
+        bool validated = true;
+    };
+
+    using Result = HandlerReturnType<Output>;
+
+    BaseServerInfoHandler(
+        std::shared_ptr<BackendInterface> const& backend,
+        std::shared_ptr<SubscriptionManagerType> const& subscriptions,
+        std::shared_ptr<ETLLoadBalancerType> const& balancer,
+        std::shared_ptr<ReportingETLType const> const& etl,
+        CountersType const& counters)
+        : backend_(backend)
+        , subscriptions_(subscriptions)
+        , balancer_(balancer)
+        , etl_(etl)
+        , counters_(std::cref(counters))
+    {
+    }
+
+    Result
+    process(Context const& ctx) const
+    {
+        using namespace RPC;
+        using namespace std::chrono;
+
+        auto const range = backend_->fetchLedgerRange();
```

Comment:
I understand it is from old code. We can remove this check , should be handled in RPC framework

Reply:
Turns out there is no such place yet as we did not integrate the new framework into clio yet. So this will have to be done later. Added a TODO comment for that.

Reply:
And tbh. i'm not even sure it should be done in framework level - not all RPC handlers will need range/db access. This is not parsing of the range from input, it's the actual range fetching code.. so we will see later how we handle it.